src/signtool/IBM_Container.cpp

```diff
@@ -0,0 +1,1767 @@
+    {
+        uint32_t value = 0;
+
+        value = (data[3] | (data[2] << 8) | (data[1] << 16) | (data[0] << 24));
+
+        return value;
+    }
+
+    uint64_t getUint64( const uint8_t *data )
+    {
+        uint64_t value = 0;
+
+        value = (            data[7]        | ((uint16_t)data[6] << 8)  | 
+                  ((uint32_t)data[5] << 16) | ((uint32_t)data[4] << 24) |
+                  ((uint64_t)data[3] << 32) | ((uint64_t)data[2] << 40) |
+                  ((uint64_t)data[1] << 48) | ((uint64_t)data[0] << 56) );
+
+        return value;
+    }
+
+    void printHexBytes( const uint8_t* bytes, size_t size, size_t leadSpace )
+    {
+        std::vector<uint8_t> byteArray;
+
+        byteArray.assign(bytes, bytes+size);
+
+        IBM_HexString hStr( byteArray );
+
+        hStr.setWidth(36);
+        hStr.setLeadSpace(leadSpace);
+
+        std::cout << hStr;
+    }
+
+  
+    const char* GetHashAlgoName( int p_hashAlgoNum )
+    {
+        static std::map<int, std::string> s_hashAlgoNames;
+
+        if (s_hashAlgoNames.empty() == 0)
+        {
+            s_hashAlgoNames[HASH_ALG_SHA512] = g_HASH_ALGO_SHA512_NAME;
+        }
+
+        std::string hashAlgoName = "Unknown";
+
+        std::map<int, std::string>::const_iterator itr = s_hashAlgoNames.find(p_hashAlgoNum);
+        if (itr != s_hashAlgoNames.end())
+        {
+            hashAlgoName = itr->second.c_str();
+        }
+
+        return hashAlgoName.c_str();
+    };
+
+  
+    const char* GetSignAlgoName( int p_signAlgoNum )
+    {
+        static std::map<int, std::string> s_signAlgoNames;
+
+        if (s_signAlgoNames.empty() == 0)
+        {
+            s_signAlgoNames[SIG_ALG_ECDSA521] = g_SIGN_ALGO_ECDSA521_NAME;
+        }
+
+        std::string signAlgoName = "Unknown";
+
+        std::map<int, std::string>::const_iterator itr = s_signAlgoNames.find((int) p_signAlgoNum);
+        if (itr != s_signAlgoNames.end())
+        {
+            signAlgoName = itr->second.c_str();
+        }
+
+        return signAlgoName.c_str();
+    };
+
+    
+    void ReadPublicKeyFromFile( const std::string&    p_mode,
+                                const std::string&    p_keyFileName,
+                                std::vector<uint8_t>& p_buffer )
+    {
+        IBM_Utils* pUtils = IBM_Utils::get();
+        THROW_EXCEPTION(pUtils == NULL);
+
+        if (p_mode == IBM_Utils::g_MODE_PRODUCTION)
+        {
+            pUtils->ReadFromFile( p_keyFileName,
+                                  p_buffer,
+                                  ECDSA521_KEY_SIZE + 1 ); // since keyfile is 133 bytes with 0x04
+                                                           // at the begining
+
+            if (p_buffer[0] != g_UNCOMPRESSED_PUBKEY_FORMAT)
+            {
+                std::stringstream ss;
+                ss << "File <" 
+                   << p_keyFileName 
+                   << "> is not a vaid p521 public key file"
+                   << std::endl;
```

Comment:
Assuming this is an error that would be printed to a user, the way you're constructing that string means you cannot translate it.